components/merkledb/src/group.rs

```diff
@@ -142,4 +207,35 @@ mod tests {
         // The next line fails to compile because `Snapshot` cannot be written to:
         // group.get(&3).push("quux".to_owned());
     }
+
+    #[test]
+    fn group_iter() {
```

Comment:
Please add a test that the iterator does not include longer index names, e.g., `"group_group_"` or `("group_group_", &0_u8)` and names situated after the group, e.g., `"unrelated"`.